### PR TITLE
Griddle QoL changes + Spatula

### PR DESCRIPTION
**Pull request by FunnyLegJiggle**
~~**I haven't finished in_hand sprites and the current sprite is "okay" but the code is done.**~~ Sprite and inhands are now done.
~~Autolathe costs aren't really final since I didn't bother looking at how they work and I didn't want an infinite mat loop.~~ Also done.

## About The Pull Request
Adds the spatula: A tool that can push all food items from a griddle onto a tray, or the first acceptable food item onto a plate in the offhand.
Griddles now cook items thrown onto them.
Makes the tray click functionality for griddles the same as drag n drop (i.e. dump contents)

## Why It's Good For The Game
Griddles were mildly annoying to add items to, and more so to get them off. Hopefully this PR changes this.

## Changelog

:cl:
add: Added the spatula, a utensil for the griddle. You can find them in the cook's vending machine or print them at cargo. It will dump all possible items onto a tray in your off-hand, or take one item off to put onto a plate in your off-hand.
qol: You can now cook things by throwing them onto the griddle.
qol: Clicking onto a griddle with a tray of food now dumps all food onto it, the same as dragging and dropping it.
/:cl:

### CODE REVIEW

File: code/modules/food_and_drinks/kitchen_machinery/griddle.dm [hunk 1]
```diff
@@ -29,9 +29,11 @@
 		variant = rand(1,3)
 	RegisterSignal(src, COMSIG_ATOM_EXPOSE_REAGENT, .proc/on_expose_reagent)
 	RegisterSignal(src, COMSIG_STORAGE_DUMP_CONTENT, .proc/on_storage_dump)
+	RegisterSignal(get_turf(src), COMSIG_ATOM_HITBY, .proc/AddThrownItemToGrill)
 
 /obj/machinery/griddle/Destroy()
 	QDEL_NULL(grill_loop)
+	UnregisterSignal(get_turf(src), COMSIG_ATOM_HITBY)
```
Comment (Kapu1178) [inline code/modules/food_and_drinks/kitchen_machinery/griddle.dm:36]: signals are cleaned up implicitly

File: code/modules/food_and_drinks/kitchen_machinery/griddle.dm [hunk 2]
```diff
@@ -90,8 +121,21 @@
 	update_appearance()
 	update_grill_audio()
 
+/obj/machinery/griddle/Moved(atom/old_loc, movement_dir, forced, list/old_locs, momentum_change)
+	UnregisterSignal(get_turf(old_loc), COMSIG_ATOM_HITBY)
+	RegisterSignal(get_turf(src), COMSIG_ATOM_HITBY, .proc/AddThrownItemToGrill)
+	. = ..()
```
Comment (Kapu1178) [inline code/modules/food_and_drinks/kitchen_machinery/griddle.dm:127]: These need to be null checked

File: code/modules/food_and_drinks/plate.dm [hunk 3]
```diff
@@ -60,6 +57,15 @@
 	SIGNAL_HANDLER
 	ItemRemovedFromPlate(moved_item)
 
+/obj/item/plate/proc/if_can_move(obj/item/I, mob/user)
```
Comment (Kapu1178) [inline code/modules/food_and_drinks/plate.dm:60]: I'd prefer this proc be named something like `can_hold_item()` or `can_accept_item()`

**Review by Kapu1178 — CHANGES_REQUESTED** [review]

File: code/modules/food_and_drinks/kitchen_machinery/griddle.dm (hunk 2 above)
Comment (FunnyLegJiggle) [reply code/modules/food_and_drinks/kitchen_machinery/griddle.dm:127]: if(.) or the the signals on the turfs existing?

File: code/modules/research/designs/autolathe_designs.dm [hunk 4]
```diff
@@ -316,6 +316,14 @@
 	category = list( DCAT_DINNERWARE)
 	mapload_design_flags = DESIGN_FAB_SERVICE | DESIGN_FAB_OMNI
 
+/datum/design/spatula
+	name = "Spatula"
+	id = "spatula"
+	build_type = AUTOLATHE
+	materials = list(/datum/material/iron = 80, /datum/material/plastic = 40)
+	build_path = /obj/item/kitchen/spatula
+	category = list("initial","Dinnerware")
```
Comment (Kapu1178) [inline code/modules/research/designs/autolathe_designs.dm:325]:
```suggestion
	category = list(DCAT_DINNERWARE)
	mapload_design_flags = DESIGN_FAB_SERVICE | DESIGN_FAB_OMNI
```

**Pull request merged**